LICENSE.AGPL

```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```

Comment:
Okay there has been a mixed up of problems. @bford talks about this problem which we discusse in the meeting:
> If the problem you see there is that all the protobuf protocols currently supported are "wired together" via nested interlocking message/struct definitions using only small-integer codes to distinguish them

@Liamsi is talking about the problem of having many different validations for BFT-CoSi, CoSi alike protocols without the need of coding them from scratch each time, but simply have this pluggable mechanism for a validation function. 
Both problems seems disjoint to me so to answer your last comment, I don't think using the extension field  will solve the second problem. Indeed scripting language for that would be nice. Such an idea popped off months ago but since we did not need it, we focused on more urgent things at the time, we need to see if that makes more sense now...